raiden/network/transport/matrix/transport.py

```diff
@@ -843,6 +844,7 @@ def _handle_message(self, room, event) -> bool:
         return True
 
     def _receive_delivered(self, delivered: Delivered):
+        assert delivered.sender is not None, MYPY_ANNOTATION
```

Comment:
@ulope Are these assert (this + 2 below) safe or are there cases where the received message has no sender?

Reply:
Since `.sender` is a property that recovers the sender address from the signature this can never be `None` for valid messages.

However since we also have to deal with possible invalid messages this can't be an assert here.
And it's actually is a bug that this doesn't check if there is a sender.

In `_receive_messge` below we check that a sender is present (i.e. the signature is valid).

Reply:
@ulope please, consider making it impossible to instantiate a message without a signature, this would make the question unnecessary and remove a few sources of bugs. (The consequence is that the signing would be decoupled from the message class).